### ISSUE TITLE
[com.driveby] Map shows incorrect position

### ISSUE THREAD

**Issue by starfunkel** [microg/GmsCore#1517]
**Affected app**
Name: MILES Carsharing & Vans
Package id: com.driveby.app

**Describe the bug**
When starting the app, the map doesn't focus /zoom to the correct position. The position varies from time to time. The app registers this false position as the current location. Therefore it's not possible to book a car in the vicinity because the app only allows to book cars in a certain range.  Despite that, The blue positional arrow indicator on the map displays the correct location.You can navigate to it. As soon as you press the localization button the app zooms to the false position. 

Sometime (mostly but not always; after a reinstall) the app displays the correct position for about 1 second and crashes). Sometimes this behavior can be reproduced by deleting cache and storage.

You don't have to have an account or have to be logged in into the app to reproduce this.

**To Reproduce**

1. Enable Location Service
2. Open the app
3. Wait for localization

**Expected behavior**
App should center on the  blue  positional indicator.

**Screenshots**
I can provide a video as well.

Initial view when the app starts
![1](https://user-images.githubusercontent.com/42552890/125514770-71b24cb7-2ace-4ca4-8db4-ce2126b8963f.png)

View after the app zooms to the false location
![2](https://user-images.githubusercontent.com/42552890/125514815-d7b2175e-2fbe-4b29-85a0-3f109725f447.png)

I zoomed out while centered the false location. You can see its off by approx. 4 kilometers.
![3](https://user-images.githubusercontent.com/42552890/125514820-d27dae94-6ba7-4f85-b80a-ff3b458f5b59.png)

**System**
Android Version: 11
Custom ROM: LineageOS 18.1

**microG**
microG Core version: 0.2.21.212158
microG Self-Check results: All ticked

**Additional context**
App signiture:

SHA1: 9464329e383738cc5eee589ef25f874abd69e1d7

Logcat output:

adb logcat | grep com.driveby.app

```` 
07-13 20:45:37.533  1650  3352 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.driveby.app/com.zoontek.rnbootsplash.RNBootSplashActivity} from uid 10141
07-13 20:45:37.594  1650  1697 I ActivityManager: Start proc 28302:com.driveby.app/u0a144 for pre-top-activity {com.driveby.app/com.zoontek.rnbootsplash.RNBootSplashActivity}
07-13 20:45:37.893 28302 28302 D SoLoader: adding application source: com.facebook.soloader.DirectorySoSource[root = /data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64 flags = 0]
07-13 20:45:37.894 28302 28302 D SoLoader: adding backup source from : com.facebook.soloader.ApkSoSource[root = /data/user/0/com.driveby.app/lib-main flags = 1]
07-13 20:45:37.894 28302 28302 D SoLoader: Preparing SO source: com.facebook.soloader.DirectorySoSource[root = /data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64 flags = 0]
07-13 20:45:37.895 28302 28302 D SoLoader: Preparing SO source: com.facebook.soloader.ApkSoSource[root = /data/user/0/com.driveby.app/lib-main flags = 1]
07-13 20:45:37.898 28302 28302 V fb-UnpackingSoSource: locked dso store /data/user/0/com.driveby.app/lib-main
07-13 20:45:37.903 28302 28302 I fb-UnpackingSoSource: dso store is up-to-date: /data/user/0/com.driveby.app/lib-main
07-13 20:45:37.903 28302 28302 V fb-UnpackingSoSource: releasing dso store lock for /data/user/0/com.driveby.app/lib-main
07-13 20:45:37.954  1650 31530 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cmp=com.driveby.app/.MainActivity} from uid 10144
07-13 20:45:37.998 28302 28355 I TetheringManager: registerTetheringEventCallback:com.driveby.app
07-13 20:45:37.999 28302 28302 D SoLoader: libjscexecutor.so not found on /data/user/0/com.driveby.app/lib-main
07-13 20:45:37.999 28302 28302 D SoLoader: libjscexecutor.so found on /data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64
07-13 20:45:38.003  1650  3353 D ConnectivityService: requestNetwork for uid/pid:10144/28302 NetworkRequest [ TRACK_DEFAULT id=268, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ]
07-13 20:45:38.004  1650  2128 D WifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=268, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
07-13 20:45:38.004  1650  2128 D UntrustedWifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=268, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
07-13 20:45:38.005  1650  2216 D Ethernet: got request NetworkRequest [ TRACK_DEFAULT id=268, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
07-13 20:45:38.005  2406  2406 D PhoneSwitcherNetworkRequstListener: got request NetworkRequest [ TRACK_DEFAULT id=268, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
07-13 20:45:38.006 28302 28364 I FA      :   adb shell setprop debug.firebase.analytics.app com.driveby.app
07-13 20:45:38.098 28302 28385 D SoLoader: libreactnativejni.so not found on /data/user/0/com.driveby.app/lib-main
07-13 20:45:38.098 28302 28385 D SoLoader: libreactnativejni.so found on /data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64
07-13 20:45:38.101 28302 28385 D SoLoader: libfb.so not found on /data/user/0/com.driveby.app/lib-main
07-13 20:45:38.101 28302 28385 D SoLoader: libfb.so found on /data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64
07-13 20:45:38.102 28302 28385 D SoLoader: libfb.so not found on /data/user/0/com.driveby.app/lib-main
07-13 20:45:38.102 28302 28385 D SoLoader: libfb.so found on /data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64
07-13 20:45:38.103 28302 28385 I com.driveby.ap: Thread[42,tid=28385,Native,Thread*=0x743f82a2e0,peer=0x132bc2e8,"create_react_context"] recursive attempt to load library "/data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64/libfb.so"
07-13 20:45:38.212  3590 13731 D GmsMeasureBrokerSvc: bound by: GetServiceRequest{serviceId=MEASUREMENT, gmsVersion=12451000, packageName='com.driveby.app', extras=Bundle[{}]}
07-13 20:45:38.219  3590 13731 D GmsMeasureSvcImpl: f4: com.driveby.app
07-13 20:45:38.236  1650  1695 I ActivityTaskManager: Displayed com.driveby.app/.MainActivity: +683ms
07-13 20:45:38.257  3590 13731 D GmsMeasureSvcImpl: f1: com.driveby.app
07-13 20:45:38.258  3590 11349 D GmsMeasureSvcImpl: f11: com.driveby.app
07-13 20:45:38.287 28302 28393 D SoLoader: libyoga.so not found on /data/user/0/com.driveby.app/lib-main
07-13 20:45:38.287 28302 28393 D SoLoader: libyoga.so found on /data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64
07-13 20:45:38.299  1650 31530 D ConnectivityService: requestNetwork for uid/pid:10144/28302 NetworkRequest [ TRACK_DEFAULT id=269, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ]
07-13 20:45:38.301  1650  2128 D WifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=269, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
07-13 20:45:38.301  1650  2128 D UntrustedWifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=269, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
07-13 20:45:38.301  1650  2216 D Ethernet: got request NetworkRequest [ TRACK_DEFAULT id=269, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
07-13 20:45:38.302  2406  2406 D PhoneSwitcherNetworkRequstListener: got request NetworkRequest [ TRACK_DEFAULT id=269, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
07-13 20:45:38.845 28302 28392 D SoLoader: libreactnativeblob.so not found on /data/user/0/com.driveby.app/lib-main
07-13 20:45:38.845 28302 28392 D SoLoader: libreactnativeblob.so found on /data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64
07-13 20:45:39.651 28302 28302 W pool-20-thread-: type=1400 audit(0.0:4786): avc: denied { read } for uid=10144 name="version" dev="proc" ino=4026532085 scontext=u:r:untrusted_app_29:s0:c144,c256,c512,c768 tcontext=u:object_r:proc_version:s0 tclass=file permissive=0 app=com.driveby.app
07-13 20:45:39.943 28302 28416 D SoLoader: libimagepipeline.so not found on /data/user/0/com.driveby.app/lib-main
07-13 20:45:39.943 28302 28416 D SoLoader: libimagepipeline.so found on /data/app/~~yVwQmA7SzhcdHYwFO_1qZg==/com.driveby.app-aT-TwOFcKNw1tpmAUjyP2A==/lib/arm64
07-13 20:45:41.745  3590 11349 D GmsMeasureSvcImpl: f1: com.driveby.app
07-13 20:45:41.945 28302 28302 D GmsMultiArchLoader: Loading mapbox-gl from /data/user/0/com.driveby.app/cache/.gmscore/lib/arm64-v8a/libmapbox-gl.so
07-13 20:45:41.955 28302 28302 W com.driveby.app: type=1400 audit(0.0:4787): avc: granted { execute } for uid=10144 path="/data/user/0/com.driveby.app/cache/.gmscore/lib/arm64-v8a/libmapbox-gl.so" dev="mmcblk0p70" ino=3194973 scontext=u:r:untrusted_app_29:s0:c144,c256,c512,c768 tcontext=u:object_r:app_data_file:s0:c144,c256,c512,c768 tclass=file app=com.driveby.app
07-13 20:45:42.183   823   823 E SurfaceFlinger: [SurfaceView - com.driveby.app/com.driveby.app.MainActivity#0] rejecting buffer: bufWidth=1080, bufHeight=2034, front.active_legacy.{w=1080, h=2033}
07-13 20:45:42.194  3590 13731 D GmsMeasureSvcImpl: f1: com.driveby.app
07-13 20:45:42.196  3590 11349 D GmsLocManagerSvc: bound by: GetServiceRequest{serviceId=LOCATION_MANAGER, gmsVersion=12451000, packageName='com.driveby.app', extras=Bundle[{client_name=locationServices}]}
07-13 20:45:43.280  3590 11349 V ULocation: onSetRequest: ProviderRequest[interval=+1s0ms] by WorkSource{10144 com.driveby.app}
07-13 20:45:43.370  3590 11349 V ULocation: onSetRequest: ProviderRequest[interval=+1s0ms] by WorkSource{1000 android, 10144 com.driveby.app}
07-13 20:45:53.206 28302 28302 I chatty  : uid=10144(com.driveby.app) identical 28 lines
07-13 20:45:53.365  3590 11349 V ULocation: onSetRequest: ProviderRequest[interval=+1s0ms] by WorkSource{10144 com.driveby.app}
07-13 20:45:54.302 28302 28302 I chatty  : uid=10144(com.driveby.app) identical 31 lines
07-13 20:45:56.005 28302 28302 I chatty  : uid=10144(com.driveby.app) identical 73 lines
07-13 20:45:56.907 28302 28302 I chatty  : uid=10144(com.driveby.app) identical 12 lines
07-13 20:45:57.868 28302 28302 I chatty  : uid=10144(com.driveby.app) identical 30 lines
07-13 20:46:20.455  3590 11349 D GmsMeasureBrokerSvc: bound by: GetServiceRequest{serviceId=MEASUREMENT, gmsVersion=12451000, packageName='com.driveby.app', extras=Bundle[{}]}
07-13 20:46:20.517  3590 11349 D GmsMeasureSvcImpl: f1: com.driveby.app
07-13 20:46:21.657  1650  1690 I ActivityManager: Killing 28302:com.driveby.app/u0a144 (adj 905): remove task
07-13 20:46:21.852  1650  2177 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ TRACK_DEFAULT id=268, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ], android.os.BinderProxy@572da06)
07-13 20:46:21.852  1650  2177 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ TRACK_DEFAULT id=269, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ], android.os.BinderProxy@6f8d8c7)
07-13 20:46:21.852  1650  2137 D ConnectivityService: releasing NetworkRequest [ TRACK_DEFAULT id=268, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] (release request)
07-13 20:46:21.853  1650  2137 D ConnectivityService: releasing NetworkRequest [ TRACK_DEFAULT id=269, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10144 AdministratorUids: [] RequestorUid: 10144 RequestorPackageName: com.driveby.app] ] (release request)
07-13 20:46:21.856  1650  3352 I WindowManager: WIN DEATH: Window{62c63c9 u0 com.driveby.app/com.driveby.app.MainActivity}
07-13 20:46:21.856  1650  3352 W InputDispatcher: Attempted to unregister already unregistered input channel '62c63c9 com.driveby.app/com.driveby.app.MainActivity (server)'
```` 
Edit 1: Inserted Code block for readability

**Comment by ntfc** [comment]
Lucky you that you are off by "only" 4km, in my case it is off by 484km :sweat_smile: 

I see some cancel request to mapbox's API, but I am not sure if that could be the issue since I can access these URLs from the device:
```
07-26 21:32:17.939   687  1058 D Mbgl-HttpRequest: [HTTP] Cancel request https://api.mapbox.com/v4/mapbox.mapbox-streets-v8,mapbox.mapbox-terrain-v2/14/8496/5468.vector.pbf?access_token=ACCESS_TOKEN&sku=SKU
07-26 21:32:17.941   687 28752 W Mbgl-HttpRequest: Request failed due to a permanent error: Canceled 
07-26 21:32:18.023   687  1058 D Mbgl-HttpRequest: [HTTP] Cancel request https://api.mapbox.com/v4/mapbox.mapbox-streets-v8,mapbox.mapbox-terrain-v2/15/16996/10937.vector.pbf?access_token=ACCESS_TOKEN&sku=SKU
07-26 21:32:18.392   687 28752 W Mbgl-HttpRequest: Request failed due to a permanent error: Canceled
```

**Comment by ale5000-git** [comment]
Could you please update microG GmsCore to v0.2.24.214816 and IchnaeaNlpBackend to v1.5.0 and post a new logcat with them?

**Comment by mguentner** [comment]
`adb logcat | grep `

```
05-05 09:19:39.182  1087  2492 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 pkg=com.driveby.app cmp=com.driveby.app/com.zoontek.rnbootsplash.RNBootSplashActivity} from uid 1000
05-05 09:19:39.209  1087  1114 I ActivityManager: Start proc 29157:com.driveby.app/u0a302 for pre-top-activity {com.driveby.app/com.zoontek.rnbootsplash.RNBootSplashActivity}
05-05 09:19:39.247 29157 25152 I TetheringManager: registerTetheringEventCallback:com.driveby.app
05-05 09:19:39.279 29157 29157 D SoLoader: adding application source: com.facebook.soloader.DirectorySoSource[root = /data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64 flags = 0]
05-05 09:19:39.279 29157 29157 D SoLoader: adding backup source from : com.facebook.soloader.ApkSoSource[root = /data/data/com.driveby.app/lib-main flags = 1]
05-05 09:19:39.280 29157 29157 D SoLoader: Preparing SO source: com.facebook.soloader.DirectorySoSource[root = /data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64 flags = 0]
05-05 09:19:39.280 29157 29157 D SoLoader: Preparing SO source: com.facebook.soloader.ApkSoSource[root = /data/data/com.driveby.app/lib-main flags = 1]
05-05 09:19:39.280 29157 29157 V fb-UnpackingSoSource: locked dso store /data/user/0/com.driveby.app/lib-main
05-05 09:19:39.281 29157 29157 I fb-UnpackingSoSource: dso store is up-to-date: /data/user/0/com.driveby.app/lib-main
05-05 09:19:39.281 29157 29157 V fb-UnpackingSoSource: releasing dso store lock for /data/user/0/com.driveby.app/lib-main
05-05 09:19:39.291 29157 25166 I FA      :   adb shell setprop debug.firebase.analytics.app com.driveby.app
05-05 09:19:39.311  1087 28630 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 pkg=com.driveby.app cmp=com.driveby.app/.MainActivity} from uid 10302
05-05 09:19:39.319  1087  2492 D ConnectivityService: requestNetwork for uid/pid:10302/29157 NetworkRequest [ TRACK_DEFAULT id=465, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ]
05-05 09:19:39.320  1087  1556 D WifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=465, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
05-05 09:19:39.320  1087  1556 D UntrustedWifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=465, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
05-05 09:19:39.320  1087  1726 D Ethernet: got request NetworkRequest [ TRACK_DEFAULT id=465, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
05-05 09:19:39.321  2234  2234 D PhoneSwitcherNetworkRequstListener: got request NetworkRequest [ TRACK_DEFAULT id=465, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
05-05 09:19:39.328 21765 21765 D GmsGcmRegister: handleMessage: package=com.driveby.app what=1 id=1
05-05 09:19:39.338 29157 29157 D SoLoader: libjscexecutor.so not found on /data/data/com.driveby.app/lib-main
05-05 09:19:39.338 29157 29157 D SoLoader: libjscexecutor.so found on /data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64
05-05 09:19:39.351 29157 25224 D SoLoader: libreactnativejni.so not found on /data/data/com.driveby.app/lib-main
05-05 09:19:39.351 29157 25224 D SoLoader: libreactnativejni.so found on /data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64
05-05 09:19:39.354 29157 25224 D SoLoader: libfb.so not found on /data/data/com.driveby.app/lib-main
05-05 09:19:39.354 29157 25224 D SoLoader: libfb.so found on /data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64
05-05 09:19:39.355 29157 25224 D SoLoader: libfb.so not found on /data/data/com.driveby.app/lib-main
05-05 09:19:39.355 29157 25224 D SoLoader: libfb.so found on /data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64
05-05 09:19:39.355 29157 25224 I com.driveby.ap: Thread[41,tid=25224,Native,Thread*=0x781fa072e0,peer=0x136c1aa8,"create_react_context"] recursive attempt to load library "/data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64/libfb.so"
05-05 09:19:39.400  3864  4167 D MeasurementService: bound by: GetServiceRequest{serviceId=MEASUREMENT, gmsVersion=12451000, packageName='com.driveby.app', extras=Bundle[{}]}
05-05 09:19:39.416  1087  1112 I ActivityTaskManager: Displayed com.driveby.app/.MainActivity: +220ms
05-05 09:19:39.439 29157 25248 I cr_VariationsUtils: Failed reading seed file "/data/user/0/com.driveby.app/app_webview/variations_seed_new"
05-05 09:19:39.503  1087  1114 I ActivityManager: Start proc 25250:com.android.webview:sandboxed_process0:org.chromium.content.app.SandboxedProcessService0:0/u0i6 for  {com.driveby.app/org.chromium.content.app.SandboxedProcessService0:0}
05-05 09:19:39.532 29157 25241 D SoLoader: libyoga.so not found on /data/data/com.driveby.app/lib-main
05-05 09:19:39.532 29157 25241 D SoLoader: libyoga.so found on /data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64
05-05 09:19:39.546  1087  3631 D ConnectivityService: requestNetwork for uid/pid:10302/29157 NetworkRequest [ TRACK_DEFAULT id=466, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ]
05-05 09:19:39.547  1087  1556 D WifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=466, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
05-05 09:19:39.547  1087  1556 D UntrustedWifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=466, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
05-05 09:19:39.547  1087  1726 D Ethernet: got request NetworkRequest [ TRACK_DEFAULT id=466, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
05-05 09:19:39.547  2234  2234 D PhoneSwitcherNetworkRequstListener: got request NetworkRequest [ TRACK_DEFAULT id=466, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10302 AdministratorUids: [] RequestorUid: 10302 RequestorPackageName: com.driveby.app] ] with score 60 and providerId 5
05-05 09:19:39.603  3864  4167 D MeasurementService: f4() for AppMetadata[com.driveby.app]
05-05 09:19:39.642  3864  4167 D MeasurementService: f1(com.google.android.gms.measurement.internal.EventParcel@4c62b9) for AppMetadata[com.driveby.app]
05-05 09:19:39.647  3864  4167 D MeasurementService: f11() for AppMetadata[com.driveby.app]
05-05 09:19:39.650  3864  4167 D MeasurementService: setDefaultEventParameters(Bundle[EMPTY_PARCEL]) for AppMetadata[com.driveby.app]
05-05 09:19:39.656  3864  4167 D MeasurementService: f10(-7127648769854425532, null, MainActivity, com.driveby.app)
05-05 09:19:39.974 29157 25238 D SoLoader: libreactnativeblob.so not found on /data/data/com.driveby.app/lib-main
05-05 09:19:39.974 29157 25238 D SoLoader: libreactnativeblob.so found on /data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64
05-05 09:19:40.671 29157 29157 W pool-24-thread-: type=1400 audit(0.0:625): avc: denied { read } for name="version" dev="proc" ino=4026532092 scontext=u:r:untrusted_app:s0:c46,c257,c512,c768 tcontext=u:object_r:proc_version:s0 tclass=file permissive=0 app=com.driveby.app
05-05 09:19:40.858 29157 25331 D SoLoader: libimagepipeline.so not found on /data/data/com.driveby.app/lib-main
05-05 09:19:40.858 29157 25331 D SoLoader: libimagepipeline.so found on /data/app/~~C2IIhmiKv-x6l-p1H76gZA==/com.driveby.app-OyieV3ojHT2T6MqAYnHaBA==/lib/arm64
05-05 09:19:41.492 29157 29157 D GmsMultiArchLoader: Loading mapbox-gl from /data/user/0/com.driveby.app/cache/.gmscore/lib/arm64-v8a/libmapbox-gl.so
05-05 09:19:41.481 29157 29157 W com.driveby.app: type=1400 audit(0.0:626): avc: granted { execute } for path="/data/data/com.driveby.app/cache/.gmscore/lib/arm64-v8a/libmapbox-gl.so" dev="sda17" ino=2044195 scontext=u:r:untrusted_app:s0:c46,c257,c512,c768 tcontext=u:object_r:app_data_file:s0:c46,c257,c512,c768 tclass=file app=com.driveby.app
05-05 09:19:41.536  3864  5206 D LocationManager: bound by: GetServiceRequest{serviceId=LOCATION_MANAGER, gmsVersion=12451000, packageName='com.driveby.app', extras=Bundle[{client_name=locationServices}]}
05-05 09:19:41.550  3864  5206 D MeasurementService: f1(com.google.android.gms.measurement.internal.EventParcel@cae87d6) for AppMetadata[com.driveby.app]
05-05 09:19:41.613  3864  5206 D MeasurementService: f1(com.google.android.gms.measurement.internal.EventParcel@9373a57) for AppMetadata[com.driveby.app]
05-05 09:19:42.127  3864  5206 V LocationProvider: onSetRequest: ProviderRequest[interval=+1s0ms] by WorkSource{10302 com.driveby.app}
05-05 09:19:42.142  3864  5206 V LocationProvider: onSetRequest: ProviderRequest[interval=+1s0ms] by WorkSource{1000 android, 10302 com.driveby.app}
05-05 09:19:46.618 29157 25238 I chatty  : uid=10302(com.driveby.app) mqt_js identical 2 lines
```

The location that is used by the application as the center of the map and also is used for business logic seems to come from the cell / IP location info or something similar. It is static but changes over time (days). The marker on the map however comes from GNSS and is constantly updated (live).

**Comment by tcecyk** [comment]
I can confirm, center-of-map is moved by IP geolocation info - tried 3 different ip routes. Mostly the zoom center was close to how db-ip.com placed the IP.

It's weird to rely on IP geloc info - mostly the data is not good enough. I very much guess it is a fallback.

The "pin/dot-mark" on the map was proper though - what Androids location service provides.

**Comment by frank-bxl** [comment]
I recognized some reproducible behaviour of the miles app: it shows the correct center-of-map (for about 10 seconds...) after:
- delete the app storage  
- restart the app and allow using the location service
After 10 seconds the app crashes and a restart after the crash leads to a center-of-map far far away (Frankfurt a. M. in my case).

**Comment by tommes0815** [comment]
I see the same issue :(

Is there any way to help solve this issue? Are there more logs needed?

**Comment by mar-v-in** [comment]
Fixed via 24b0bfa998a23ace0851e5181cf5f3c13db28b5c
